src/pyload/webui/app/blueprints/json_blueprint.py

```diff
@@ -31,46 +31,27 @@ def links():
         links = api.status_downloads()
         ids = []
         for link in links:
-            ids.append(link["fid"])
-
-            if link["status"] == 12:  #: downloading
-                formatted_eta = link["format_eta"]
-                formatted_speed = format.speed(link["speed"])
-                link["info"] = f"{formatted_eta} @ {formatted_speed}"
-
-            elif link["status"] == 5:  #: waiting
-                link["percent"] = 0
-                link["size"] = 0
-                link["bleft"] = 0
-                link["info"] = api._("waiting {}").format(link["format_wait"])
+            ids.append(link.fid)
+
+            if link.status == 12:  #: downloading
+                formatted_eta = link.format_eta
+                formatted_speed = format.speed(link.speed)
+                link.info = f"{formatted_eta} @ {formatted_speed}"
+
+            elif link.status == 5:  #: waiting
+                link.percent = 0
+                link.size = 0
+                link.bleft = 0
+                link.info = api._("waiting {}").format(link.format_wait)
             else:
-                link["info"] = ""
+                link.info = ""
 
         return jsonify(links=links, ids=ids)
 
     except Exception as exc:
         return jsonify(False), 500
 
 
-@bp.route("/json/packages", endpoint="packages")
-# @apiver_check
-@login_required("LIST")
-def packages():
-    api = flask.current_app.config["PYLOAD_API"]
-    try:
-        data = api.get_queue()
-
-        for package in data:
-            package["links"] = []
-            for file in api.get_package_files(package["id"]):
-                package["links"].append(api.get_file_info(file))
-
-        return jsonify(data)
-
-    except Exception:
-        return jsonify(False), 500
-
-
```

Comment:
I deleted this function because I don't think it actually worked.

it calls `get_queue()` which returns `PackageData` and then tries to access it `id` - but `PackageData` has no `id` property/key.
Even if we assume that `pid` was meant here, then it will call `api.get_package_files()` which doesn't exist.

So something was very broken here. If you think we should keep it then please let me know how to fix this method to actually work as expected.